[CI] ProfileDomainIntegTests testConcurrentActivateUpdates failing

**Build scan:**
https://gradle-enterprise.elastic.co/s/turc6aa5aqtig/tests/:x-pack:plugin:security:internalClusterTest/org.elasticsearch.xpack.security.profile.ProfileDomainIntegTests/testConcurrentActivateUpdates

**Reproduction line:**
`./gradlew ':x-pack:plugin:security:internalClusterTest' --tests "org.elasticsearch.xpack.security.profile.ProfileDomainIntegTests.testConcurrentActivateUpdates" -Dtests.seed=2CB1A2766759F141 -Dtests.locale=pl -Dtests.timezone=Europe/Dublin -Druntime.java=17`

**Applicable branches:**
main

**Reproduces locally?:**
Didn't try

**Failure history:**
https://gradle-enterprise.elastic.co/scans/tests?tests.container=org.elasticsearch.xpack.security.profile.ProfileDomainIntegTests&tests.test=testConcurrentActivateUpdates

**Failure excerpt:**
```
java.lang.AssertionError: 
Expected: <Profile[uid=u_ZStoPhTrsYcQDKq6Oi45MDetmYMt2-fGNkdwO9LxDb4_0, enabled=true, lastSynchronized=1662038493111, user=ProfileUser[username=wiNPJHcQKR, roles=[foo], realmName=Jdts, domainName=null, email=null, fullName=null], labels={}, applicationData={}, versionControl=VersionControl[primaryTerm=1, seqNo=2]]>
     but: was <Profile[uid=u_ZStoPhTrsYcQDKq6Oi45MDetmYMt2-fGNkdwO9LxDb4_0, enabled=true, lastSynchronized=1662038493110, user=ProfileUser[username=wiNPJHcQKR, roles=[foo], realmName=Jdts, domainName=null, email=null, fullName=null], labels={}, applicationData={}, versionControl=VersionControl[primaryTerm=1, seqNo=3]]>

  at __randomizedtesting.SeedInfo.seed([2CB1A2766759F141:5D5C9D00FF0D3F43]:0)
  at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
  at org.junit.Assert.assertThat(Assert.java:956)
  at org.junit.Assert.assertThat(Assert.java:923)
  at org.elasticsearch.xpack.security.profile.ProfileDomainIntegTests.testConcurrentActivateUpdates(ProfileDomainIntegTests.java:403)
  at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(NativeMethodAccessorImpl.java:-2)
  at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
  at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
  at java.lang.reflect.Method.invoke(Method.java:568)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:982)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
  at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:48)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:44)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:45)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:843)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:490)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:38)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:47)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:850)
  at java.lang.Thread.run(Thread.java:833)

```

--- a/x-pack/plugin/security/src/internalClusterTest/java/org/elasticsearch/xpack/security/profile/ProfileDomainIntegTests.java
+++ b/x-pack/plugin/security/src/internalClusterTest/java/org/elasticsearch/xpack/security/profile/ProfileDomainIntegTests.java
@@ -302,9 +302,9 @@ public class ProfileDomainIntegTests extends AbstractProfileIntegTestCase {
                     final Authentication authentication = assembleAuthentication(username, randomRealmRef());
                     final ProfileService profileService = getInstanceFromRandomNode(ProfileService.class);
                     final PlainActionFuture<Profile> future = new PlainActionFuture<>();
-                    profileService.activateProfile(authentication, future);
                     readyLatch.countDown();
                     startLatch.await();
+                    profileService.activateProfile(authentication, future);
                     try {
                         final String uid = future.actionGet().uid();
                         logger.info("created profile [{}] for authentication [{}]", uid, authentication);
@@ -373,9 +373,9 @@ public class ProfileDomainIntegTests extends AbstractProfileIntegTestCase {
             threads[i] = new Thread(() -> {
                 try {
                     final PlainActionFuture<Profile> future = new PlainActionFuture<>();
-                    profileService.activateProfile(updatedAuthentication, future);
                     readyLatch.countDown();
                     startLatch.await();
+                    profileService.activateProfile(updatedAuthentication, future);
                     final Profile updatedProfile = future.actionGet();
                     assertThat(updatedProfile.uid(), equalTo(originalProfile.uid()));
                     assertThat(updatedProfile.user().roles(), contains("foo"));
@@ -393,14 +393,18 @@ public class ProfileDomainIntegTests extends AbstractProfileIntegTestCase {
             for (Thread thread : threads) {
                 thread.join();
             }
+            assertThat(updatedProfiles, not(emptyIterable()));
+            // Find the most recent version of the updated profile document using the seqNo which is monotonically increasing
+            final Profile updatedProfile = updatedProfiles.stream()
+                .max(Comparator.comparingLong(p -> p.versionControl().seqNo()))
+                .orElseThrow();
+            // Update again, this time it should simply skip due to grace period of 30 seconds
+            final PlainActionFuture<Profile> future = new PlainActionFuture<>();
+            profileService.activateProfile(updatedAuthentication, future);
+            assertThat(future.actionGet(), equalTo(updatedProfile));
+        } else {
+            fail("Not all threads are ready after waiting");
         }
-
-        assertThat(updatedProfiles, not(emptyIterable()));
-        final Profile updatedProfile = updatedProfiles.stream().max(Comparator.comparingLong(Profile::lastSynchronized)).orElseThrow();
-        // Update again, this time it should simply skip due to grace period of 30 seconds
-        final PlainActionFuture<Profile> future = new PlainActionFuture<>();
-        profileService.activateProfile(updatedAuthentication, future);
-        assertThat(future.actionGet(), equalTo(updatedProfile));
     }
 
     public void testDifferentiator() {

